apps/dashboard/src/components/primitives/form/form.tsx

```diff
@@ -115,12 +115,22 @@ type FormMessagePureProps = React.HTMLAttributes<HTMLParagraphElement> & { hasEr
 const FormMessagePure = React.forwardRef<HTMLParagraphElement, FormMessagePureProps>(
   ({ className, children, hasError = false, icon, ...props }, _ref) => {
     return (
-      children && (
-        <Hint hasError={hasError} {...props}>
-          {icon && <HintIcon as={icon} />}
-          {children}
-        </Hint>
-      )
+      <AnimatePresence mode="wait">
+        {children && (
+          <motion.div
+            key={hasError ? 'error' : 'empty'}
+            initial={{ opacity: 0, y: -5, height: 0 }}
+            animate={{ opacity: 1, y: 0, height: 'auto' }}
+            exit={{ opacity: 0, y: -5, height: 0 }}
+            transition={{ duration: 0.3 }}
+          >
+            <Hint hasError={hasError} {...props}>
+              {icon && <HintIcon as={icon} />}
+              {children}
+            </Hint>
+          </motion.div>
+        )}
+      </AnimatePresence>
```

Comment:
moved this from the `FormMessage` to `FormMessagePure` so we can see this animation in all places where both are used; 
simplified the code a bit